plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -649,4 +654,141 @@ private function print_hidden_form_fields(): void {
 			echo '<input type="hidden" name="status" value="' . esc_attr( sanitize_text_field( wp_unslash( $_GET[ $param ] ) ) ) . '" >';
 		}
 	}
+
+	/**
+	 * Handle bulk actions.
+	 */
+	public function handle_bulk_actions() {
+		$action = $this->current_action();
+
+		if ( ! $action ) {
+			return;
+		}
+
+		check_admin_referer( 'bulk-orders' );
+
+		$redirect_to = remove_query_arg( array( 'deleted', 'ids' ), wp_get_referer() );
+		$redirect_to = add_query_arg( 'paged', $this->get_pagenum(), $redirect_to );
+
+		/**
+		 * Allows 3rd parties to modify order IDs about to be affected by a bulk action.
+		 *
+		 * @param array Array of order IDs.
+		 */
+		$ids = apply_filters( // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingSinceComment
+			'woocommerce_bulk_action_ids',
+			isset( $_REQUEST['order'] ) ? array_reverse( array_map( 'absint', $_REQUEST['order'] ) ) : array(),
+			$action,
+			'order'
+		);
+
+		if ( ! $ids ) {
+			wp_safe_redirect( $redirect_to );
+			exit;
+		}
+
+		$report_action = '';
+		$changed       = 0;
+
+		if ( 'remove_personal_data' === $action ) {
+			$report_action = 'removed_personal_data';
+			$changed       = $this->do_bulk_action_remove_personal_data( $ids );
+		} elseif ( false !== strpos( $action, 'mark_' ) ) {
+			$order_statuses = wc_get_order_statuses();
+			$new_status     = substr( $action, 5 );
+			$report_action  = 'marked_' . $new_status;
+
+			if ( isset( $order_statuses[ 'wc-' . $new_status ] ) ) {
+				$changed = $this->do_bulk_action_mark_orders( $ids, $new_status );
+			}
+		}
+
+		if ( $changed ) {
+			$redirect_to = add_query_arg(
+				array(
+					'bulk_action' => $report_action,
+					'changed'     => $changed,
+					'ids'         => implode( ',', $ids ),
+				),
+				$redirect_to
+			);
+		}
+
+		wp_safe_redirect( $redirect_to );
+		exit;
+	}
+
+	/**
+	 * Implements the "remove personal data" bulk action.
+	 *
+	 * @param array $order_ids The Order IDs.
+	 * @return int Number of orders modified.
+	 */
+	private function do_bulk_action_remove_personal_data( $order_ids ): int {
+		$changed = 0;
+
+		foreach ( $order_ids as $id ) {
+			$order = wc_get_order( $id );
+
+			if ( $order ) {
+				do_action( 'woocommerce_remove_order_personal_data', $order ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+				$changed++;
+			}
+		}
+
+		return $changed;
+	}
+
+	/**
+	 * Implements the "mark <status>" bulk action.
+	 *
+	 * @param array  $order_ids  The order IDs to change.
+	 * @param string $new_status The new order status.
+	 * @return int Number of orders modified.
+	 */
+	private function do_bulk_action_mark_orders( $order_ids, $new_status ): int {
+		$changed = 0;
+
+		// Initialize payment gateways in case order has hooked status transition actions.
+		WC()->payment_gateways();
+
+		foreach ( $order_ids as $id ) {
+			$order = wc_get_order( $id );
+			$order->update_status( $new_status, __( 'Order status changed by bulk edit:', 'woocommerce' ), true );
```

Comment:
```suggestion
			$order->update_status( $new_status, __( 'Order status changed by bulk edit.', 'woocommerce' ), true );
```

Definitely not essential we change this (apart from being super minor, I realize this is exactly what we already do with post-based orders), but the above might read a bit better, as the message we pass here and the status update string are really two separate sentences:

<table>
<thead><tr><th>From</th><th>To</th></tr></thead>
<tbody><tr>
<td><img width="274" alt="colon" src="https://user-images.githubusercontent.com/3594411/177218182-b0a6e266-8810-4b91-9a24-f42a762e4e53.png"></td>
<td><img width="272" alt="period" src="https://user-images.githubusercontent.com/3594411/177218184-fc5c38ed-231a-42fd-be18-acce7e1e3e25.png"></td>
</tr></tbody></table>